src/server/service/match.ts

```diff
@@ -325,6 +340,23 @@ export class Match {
             score: sql`${users.score} + ${player.score}`,
           })
           .where(eq(users.id, player.userId));
+
+        player.achievements
+          .filter((achievement) => {
+            return (
+              achievement === "beginnersLuck" ||
+              achievement === "realHuman" ||
+              achievement === "firstTimer" ||
+              achievement === "busterStreak"
+            );
+          })
+          .map(async (achievementId) => {
+            await dbTx.insert(userAchievements).values({
+              userId: player.userId,
+              achievementId: achievementId,
+              achievedAt: new Date(),
```

Comment:
Shouldn't this be the original date where the achievement was first unlocked?

Reply:
This is the place where the achievements are unlocked.